src/core/addonManager.js

```diff
@@ -9,8 +9,13 @@ import {
 import { addQueryParams } from 'core/utils';
 
 
-export function hasAddonManager({ navigator } = {}) {
-  return typeof window !== 'undefined' && 'mozAddonManager' in (navigator || window.navigator);
+const testHasWindow = () => typeof window !== 'undefined';
+
+export function hasAddonManager({ hasWindow = testHasWindow, navigator } = {}) {
+  if (!navigator && !hasWindow()) {
+    return undefined;
```

Comment:
So you can tell the difference between "not supported" and "we don't know if it's supported".